### ISSUE TITLE
NavigationMode not in NavigationParameters

### ISSUE THREAD

**Issue by dazinator** [PrismLibrary/Prism#1336]
﻿### Description
I am experiencing the following problem since updating to the latest nuget packages for xamarin forms.
`NavigationParameters` doesn't appear to contain a `NavigationMode` as it did before.

For my `MainPage` am using a `MasterDetail` page.
I navigate from there like so:

```csharp
 Detail = new NavigationPage((Page)Activator.CreateInstance(typeof(WorkAllocationPage)));
```

When the view model loads for the WorkAllocationPage, I have a button click handler that navigates me onwards to another `ContentPage` like so:

```csharp
                        var navParameters = new NavigationParameters();
                        navParameters.Add(CurrentVisitModelKeyName, visitModel);                    

                        DeviceService.BeginInvokeOnMainThread(async () =>
                        {                          
                            string navUrl = string.Format("/{0}"
                            , NavigationPages.GroupSelectionPage.ToString());

                            await NavigationService.NavigateAsync(new Uri(navUrl, UriKind.Relative), navParameters, animated: false);
                        });
```

When that next viewmodel loads, the following override throws on `GetNavigationMode` because there isn't one present:
```

        public override async void OnNavigatedTo(NavigationParameters parameters)
        {
            //Call Base Method
            base.OnNavigatedTo(parameters);

            NavigationMode navigationMode = parameters.GetNavigationMode();

            if (navigationMode == NavigationMode.New)
            {
                //Download 
                await DownloadAndInstallLatestVersion();
            }
        }

```

### Steps to Reproduce

I don't have a sample I can share at the moment, but perhaps the above description will be helpful.

### Expected Behavior

Should find navigation mode.

### Actual Behavior

No navigation mode in navigation parameters.

### Basic Information

- Version with issue: 7.0.0.362,
- Last known good version: 7.0.0.168-pre
- Xamarin.Forms version: 2.5.0.122203 (When last working I was on 2.5.0.121934)
- IDE: VS2017 15.5.4

**Comment by brianlagunas** [comment]
I'm going to need a sample. You're doing some strange things that may be causing the problem.  As you can see, any time you navigate in Prism it adds the parameter

https://github.com/PrismLibrary/Prism/blob/master/Source/Xamarin/Prism.Forms/Navigation/PageNavigationService.cs#L545

**Comment by dazinator** [comment]
Yeah that's fair. I haven;t had time to put together a repro for you, so i am just using a workaround of setting the nav mode myself each time I navigate for the time being. If I get time to spend on it further I'll trace a root cause and provide a repro if necessary.
